Library/Homebrew/version.rb

```diff
@@ -376,8 +376,9 @@ def self.parse(spec, detected_from_url: false)
 
   VERSION_PARSERS = [
     # date-based versioning
+    # e.g. 2023-09-28.tar.gz
     # e.g. ltopers-v2017-04-14.tar.gz
-    StemParser.new(/-v?(\d{4}-\d{2}-\d{2})/),
+    StemParser.new(/[._-]?v?(\d{4}-\d{2}-\d{2})/),
```

Comment:
```suggestion
    StemParser.new(/(^|[._-])?v?(\d{4}-\d{2}-\d{2})/),
```

So it doesn't also match 222222222-11-11? 

Or maybe this will work?

```suggestion
    StemParser.new(/(?:\b|v)(\d{4}-\d{2}-\d{2})/),
```

Reply:
Thanks for bringing this up. This is something that I'd been thinking about (after creating this PR) but hadn't gotten around to yet. The first suggestion is along the lines of what I was thinking and it worked as expected when I tested it (i.e., it continues to match what we want without the risk of matching something unexpected, like your example).